### DISCUSSION TITLE
Uncastable from request into validation error

### DISCUSSION THREAD

**Discussion by wietsewarendorff** [spatie/laravel-data#857] [Q&A]
Hi, 

Is there any way to convert an uncastable into a validation error?

I am returning an `Uncastable::create()` from a custom cast for a property, which I am injecting in a request. I expected that invalid data (hence the uncastable) would result in a validation error but it just throws an exception (ie `Argument given must be of type X, string given`.

example code to illustrate:

controller:
```
class Controller
{
    public function __invoke(Dto $dto);
}
```

dto:
```
class Dto
{
    public function __construct(public X $x);
}
````

cast:
```
class XCast implements Cast
{
    public function cast(...)
    {
        return new Uncastable::create();
    }
}
```

post a request with the following body:
```
    x: 'y'
```

I understand validation is applied before casting (which makes sense), but I would like the data to be validated before attempting the cast. This would prevent the Uncastable, which just "returns the passed value" (the string "x" in the example) at the moment.

I could add (custom) validation rules, but I think it would be better if the rules are automatically applied. Any idea, or am I missing something obvious?

Thanks in advance

**Comment by rubenvanassche** [comment]
You could just throw your own ValidationException? CreationContext has the path to set for the message

**Reply by wietsewarendorff** [reply]
thanks for taking time to look into this!

throwing an validationException dit not result in a validation error when injecting the data-object as a request paramater in my controller (which I hoped).

already have a work-around, so no longer need this.
